contracts/core/extensions/CoreIssuance.sol

```diff
@@ -204,6 +204,73 @@ contract CoreIssuance is
         }
     }
 
+    /**
+     * Function to convert Set Tokens held in the vault into underlying components in vault
+     *
+     * @param _set          The address of the Set token
+     * @param _quantity     The number of tokens to redeem
+     */
+    function redeemInVault(
+        address _set,
+        uint _quantity
+    )
+        external
+    {
+        // Declare interface variables
+        ISetToken setToken = ISetToken(_set);
+        IVault vault = IVault(state.vault);
+
+        // Verify Set was created by Core and is enabled
+        require(state.validSets[_set]);
+
+        // Validate quantity is multiple of natural unit
+        require(_quantity % setToken.naturalUnit() == 0);
+
+        // Decrement ownership of Set token in the vault
+        vault.decrementTokenOwner(
+            msg.sender,
+            _set,
+            _quantity
+        );
+
+        // Burn the Set token from the vault's amount
+        setToken.burn(
```

Comment:
This is the only statement that's different from redeem. Can we combine the two functions and have one call the other? The public redeem must be called with msg.sender.

Reply:
Well that and decrementing the sender's setToken balance in the vault. I feel like it's annoyingly different enough to have to have a full fxn. We could make everything below that a separate fxn.